pkg/adapter/v2/cloudevents_test.go

```diff
@@ -161,20 +163,23 @@ func TestNewCloudEventsClient_send(t *testing.T) {
 				t.Errorf("expected NewCloudEventsClient to return a *client, but did not")
 			}
 			innerClient := &test.TestCloudEventsClient{}
-			if tc.result != nil {
-				innerClient.Send_AppendResult(tc.result)
-			}
 			got.ceClient = innerClient
 
 			if tc.event != nil {
+				if n == "not a http result" {
```

Comment:
Curious why we don't set the type in the table directly and override it here instead? Maybe a comment explaining would help? I'm just thinking that it's a bit tricky to follow the special cases here and in L173 and L178. Meaning it's not obvious from the table tests set up what's being tested.

Reply:
@vaikas I agree with you on that. like I mentioned in the comment above there is some refactoring need because I am not sure why we have created a test case table where we define various cases and the expected results but we can't use them because when we call `send` or `request` method via test_client, it only looks at the events part. I was thinking the conditional handling should happen there if we could use the whole `resultRequest` there.